apollo-federation/src/sources/connect/validation/http/url.rs

```diff
@@ -0,0 +1,85 @@
+use std::fmt::Display;
+use std::ops::Range;
+use std::str::FromStr;
+
+use apollo_compiler::ast::Value;
+use apollo_compiler::parser::LineColumn;
+use apollo_compiler::parser::SourceMap;
+use apollo_compiler::Node;
+use url::Url;
+
+use crate::sources::connect::url_template;
+use crate::sources::connect::validation::coordinates::HttpMethodCoordinate;
+use crate::sources::connect::validation::require_value_is_str;
+use crate::sources::connect::validation::Code;
+use crate::sources::connect::validation::Message;
+use crate::sources::connect::URLTemplate;
+
+pub(crate) fn validate_template(
+    value: &Node<Value>,
+    coordinate: HttpMethodCoordinate,
+    sources: &SourceMap,
+) -> Result<URLTemplate, Message> {
+    let str_value = require_value_is_str(value, coordinate, sources)?;
+    let template =
+        URLTemplate::from_str(str_value).map_err(|url_template::Error { message, location }| {
+            Message {
+                code: Code::InvalidUrl,
+                message: format!("{coordinate} must be a valid URL template. {message}",),
```

Comment:
i've noticed a few trailing commas, is this a jetbrains formatting thing?

Reply:
No, it's a `rustfmt` doesn't get rid of them if they're left over from elsewhere thing 🙃. I'll try to hunt them down by hand.

Reply:
Well, searching globally for `,)` in our repo certain illustrates how common that limitation is. Seems like it only happens in macro invocations. I _think_ I removed all the ones I added here.